### PR TITLE
Add the 2.5.0 release of mirage

### PR DESCRIPTION
**Pull request by samoht**
Please do not merge, I'll do it myself when everything else is ready.

### CODE REVIEW

File: packages/mirage-conduit/mirage-conduit.2.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+homepage:     "https://github.com/mirage/ocaml-conduit"
+dev-repo:     "https://github.com/mirage/ocaml-conduit.git"
+bug-reports:  "https://github.com/mirage/ocaml-conduit/issues"
+tags:         "org:mirage"
+
+build:   ["ocamlfind" "query" "conduit.mirage"]
+depends: [
+  "mirage-types" {>= "2.0.0"}
+  "mirage-dns" {>= "2.0.0"}
+  "tcpip"
```
Comment (avsm) [inline packages/mirage-conduit/mirage-conduit.2.2.0/opam]: possibly good idea to have lower bound on tcp too

File: packages/mirage-http/mirage-http.2.3.0/opam [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+build:    [make]
+install: [make "install"]
+remove:  ["ocamlfind" "remove" "mirage-http"]
+depends: [
+  "ocamlfind" {build}
+  "mirage-types-lwt" {>= "2.0.0"}
+  "mirage-conduit"   {>= "2.2.0"}
+  "lwt" {>= "2.4.3"}
+  "cohttp" {>= "0.16.0" & < "0.18.0"}
+  "camlp4"
```
Comment (avsm) [inline packages/mirage-http/mirage-http.2.3.0/opam]: i think the camlp4 should actually be type_conv here, to ensure sexplib.syntax is available

File: packages/mirage-http/mirage-http.2.4.0/opam [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+build:    [make]
+install: [make "install"]
+remove:  ["ocamlfind" "remove" "mirage-http"]
+depends: [
+  "ocamlfind" {build}
+  "mirage-types-lwt" {>= "2.0.0"}
+  "mirage-conduit" {>= "2.2.0"}
+  "lwt" {>= "2.4.3"}
+  "cohttp" {>= "0.18.0"}
+  "camlp4"
```
Comment (avsm) [inline packages/mirage-http/mirage-http.2.4.0/opam]: same here -- type_conv not camlp4

File: packages/mirage/mirage.2.5.0/opam [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+depends: [
+  "ipaddr" {>= "1.0.0"}
+  "mirage-types" {>= "2.3.0"}
+  "re" {>= "1.2.1"}
+  "cmdliner" {>= "0.9.2"}
+  "lwt" {>= "2.4.3"}
+  "cstruct" {>= "1.0.1"}
+  "io-page" {>= "1.4.0"}
+  "crunch" {>= "1.2.2"}
+  "sexplib"
```
Comment (avsm) [inline packages/mirage/mirage.2.5.0/opam]: doesnt mirage use `with sexp` and so needs type_conv as well?

Comment (samoht) [inline packages/mirage/mirage.2.5.0/opam]: actually we don't use sexplib anymore, I'll remove it

File: packages/mirage-http/mirage-http.2.4.0/opam (hunk 3 above)
Comment (samoht) [inline packages/mirage-http/mirage-http.2.4.0/opam]: we don't use camlp4 anymore actually, so I'll remove the dep.

File: packages/mirage-conduit/mirage-conduit.2.2.0/opam (hunk 1 above)
Comment (samoht) [inline packages/mirage-conduit/mirage-conduit.2.2.0/opam]: good idea, I'll add a constraint to use the latest tcpip release (with all the nice fixes)

**Comment by samoht** [comment]
Travis is very buggy... all the tests are not green but it still displays a red cross ... Merging anyway.

**Pull request merged**